frontend/src/Components/ProtectedRoute.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+import { useLoading } from '../Loading/LoadingContext';
+import PropTypes from 'prop-types';
+import { API_BASE_URL } from '../config';
+
+
+const ProtectedRoute = ({ children }) => {
+const [isAuthenticated, setIsAuthenticated] = useState(false);
+const { isLoading, setIsLoading } = useLoading();
+const [error, setError] = useState(null);
+
+
+useEffect(() => {
+    const checkAuth = async () => {
+    setIsLoading(true);
+    try {
+        const response = await fetch(`${API_BASE_URL}/users/auth-check`, {
+        credentials: 'include'
+        });
+        if (response.ok) {
+        setIsAuthenticated(true);
+        } else {
+        setIsAuthenticated(false);
+        }
+    } catch (error) {
+        console.error('Auth check failed:', error);
```

Comment:
Remove this.

Reply:
Hey I would need more clarification on this? Which part do you want me to remove?

Reply:
I'm referring to the "console.error". If you are done debugging, let's clean it up.